src/cursor/cur_stat.c

```diff
@@ -383,6 +383,75 @@ __curstat_conn_init(WT_SESSION_IMPL *session, WT_CURSOR_STAT *cst)
     cst->stats_desc = __wt_stat_connection_desc;
 }
 
+/*
+ * __init_layered_constituent_stats --
+ *     For either an ingest or stable table, aggregate dhandle stats into a stat cursor.
+ */
+static int
+__init_layered_constituent_stats(WT_SESSION_IMPL *session, WT_CURSOR_STAT *cst, bool *stats_cleared)
+{
+    WT_DATA_HANDLE *dhandle;
+    WT_DECL_RET;
+
+    dhandle = session->dhandle;
+
+    if ((ret = __wt_btree_stat_init(session, cst)) == 0) {
+        /*
+         * Only clear the aggregate stats if (1) we initialized this tree's stats, and (2) another
+         * constituent call didn't do it.
+         */
+        if (!*stats_cleared)
```

Comment:
Can we just call __wt_stat_dsrc_init_single at the start of __curstat_layered_init? Why it is not OK to call __wt_stat_dsrc_init_single before we do __wt_btree_stat_init?

Reply:
The existing file stats are like this so it might be a Don question, but my understanding is that we shouldn't wipe the aggregate stats if we're not going to be able to use them (because the btree stat init failed).

In practice I think you're right because `__Wt_btree_stat_init` can only return 0, so unless @ddanderson has a reason, I'll rework it to pull the `init_single` out.

Reply:
I think that's right, it will be easier to understand if the call to init_single is at the beginning of `__curstat_layered_init`.

Reply:
Good call - fixing this has improved the code a bunch. Fixed in fb3f2aa.